### PR TITLE
Implement `timespec_get`

### PR DESCRIPTION
**Pull request by catap**

### CODE REVIEW

**Comment by catap** [comment]
@kencu may I ask you to review this one?

**Review by mascguy — APPROVED** [review]
LGTM. @kencu?

File: README.md [hunk 1]
```diff
@@ -88,6 +88,11 @@ Wrapped headers and replaced functions are:
     <td>Adds <code>clock_gettime</code> function (macOS10.11). Declares <code>asctime_r</code>, <code>ctime_r</code>, <code>gmtime_r</code>, and <code>localtime_r</code> functions that are otherwise hidden in the presence of <code>_ANSI_SOURCE</code>, <code>_POSIX_C_SOURCE</code>, or <code>_XOPEN_SOURCE</code> (OSX10.4)</td>
     <td>OSX10.4(11)</td>
   </tr>
+  <tr>
+    <td><code>time.h</code></td>
```
Comment (michaelld) [inline README.md:92]: This should be integrated into the prior entry (which is for `time.h`)

**Review by michaelld — CHANGES_REQUESTED** [review]
small change then this is good to go IMHO

File: README.md (hunk 1 above)
Comment (catap) [reply README.md]: @michaelld like this?

File: README.md [hunk 2]
```diff
@@ -85,8 +85,8 @@ Wrapped headers and replaced functions are:
   </tr>
   <tr>
     <td><code>time.h</code></td>
-    <td>Adds <code>clock_gettime</code> function (macOS10.11). Declares <code>asctime_r</code>, <code>ctime_r</code>, <code>gmtime_r</code>, and <code>localtime_r</code> functions that are otherwise hidden in the presence of <code>_ANSI_SOURCE</code>, <code>_POSIX_C_SOURCE</code>, or <code>_XOPEN_SOURCE</code> (OSX10.4)</td>
-    <td>OSX10.4(11)</td>
+    <td>Adds <code>clock_gettime</code>(macOS10.11) and <code>timespec_get</code>(macOS10.14) functions. Declares <code>TIME_UTC</code> defines (macOS10.14), <code>asctime_r</code>, <code>ctime_r</code>, <code>gmtime_r</code>, and <code>localtime_r</code> functions that are otherwise hidden in the presence of <code>_ANSI_SOURCE</code>, <code>_POSIX_C_SOURCE</code>, or <code>_XOPEN_SOURCE</code> (OSX10.4)</td>
```
Comment (michaelld) [inline README.md:88]: How about this:
```
    <td>Adds functions <code>clock_gettime</code>(macOS10.11) and <code>timespec_get</code>(macOS10.14). Defines <code>TIME_UTC</code> (macOS10.14). Declared <code>asctime_r</code>, ...
```

**Review by michaelld — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (catap) [reply README.md:88]: much better. Thanks.

File: README.md [hunk 3]
```diff
@@ -85,8 +85,8 @@ Wrapped headers and replaced functions are:
   </tr>
   <tr>
     <td><code>time.h</code></td>
-    <td>Adds <code>clock_gettime</code> function (macOS10.11). Declares <code>asctime_r</code>, <code>ctime_r</code>, <code>gmtime_r</code>, and <code>localtime_r</code> functions that are otherwise hidden in the presence of <code>_ANSI_SOURCE</code>, <code>_POSIX_C_SOURCE</code>, or <code>_XOPEN_SOURCE</code> (OSX10.4)</td>
-    <td>OSX10.4(11)</td>
+    <td>Adds functions <code>clock_gettime</code>(macOS10.11) and <code>timespec_get</code>(macOS10.14). Defines <code>TIME_UTC</code> (macOS10.14). Declared <code>asctime_r</code>, <code>ctime_r</code>, <code>gmtime_r</code>, and <code>localtime_r</code> functions that are otherwise hidden in the presence of <code>_ANSI_SOURCE</code>, <code>_POSIX_C_SOURCE</code>, or <code>_XOPEN_SOURCE</code> (OSX10.4)</td>
```
Comment (michaelld) [inline README.md:88]: "Declared" -> "Declares" ... then I'm  really done! Thanks for all these changes!

**Review by michaelld — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (catap) [reply README.md]: here it is

**Comment by michaelld** [comment]
Seeing if @kencu is around for a review too ...

**Comment by kencu** [comment]
As far as i can see it seems good:

```
% cat timetest.c
#include <time.h>
#include <stdio.h>

int mytimespec_get(struct timespec *ts, int base)
{
  switch (base) {
    case TIME_UTC:
      if (clock_gettime(CLOCK_REALTIME, ts) == -1)
        return 0;

      return base;

    default:
      return 0;
  }
}



int
main(void) {
   int error;
   struct timespec mytime;

   error = timespec_get( &mytime, TIME_UTC);
   printf("%lld.%.9ld\n", (long long)mytime.tv_sec, mytime.tv_nsec);
   printf("error: %i\n", error);

   error = mytimespec_get( &mytime, TIME_UTC);
   printf("%lld.%.9ld\n", (long long)mytime.tv_sec, mytime.tv_nsec);
   printf("error: %i\n", error);

}

MacBookPro-2012 kentest % clang -o timetest timetest.c
MacBookPro-2012 kentest % ./timetest
1662773680.750552000
error: 1
1662773680.750585000
error: 1
MacBookPro-2012 kentest % ./timetest
1662773685.446401000
error: 1
1662773685.446434000
error: 1
MacBookPro-2012 kentest % ./timetest
1662773687.110421000
error: 1
1662773687.110455000
error: 1
```

I'm not exactly sure how the test case would be written, as you can never get the exact same result, and it's really only valid on a machine that can support mytimespec_get, but perhaps some little test that shows it works would do, ie some variation on this that prints "success" if it works:

```
#include <time.h>
#include <stdio.h>

int
main(void) {
   int error;
   struct timespec mytime;

   error = timespec_get( &mytime, TIME_UTC);
   printf("%lld.%.9ld\n", (long long)mytime.tv_sec, mytime.tv_nsec);
   printf("error: %i\n", error);

}
```

**Comment by kencu** [comment]
by the way, doing all this on the new "codeblocks-devel" port that is fun to work with for stuff like this...

**Pull request merged**